app/routes.py

```diff
@@ -1,2 +1,51 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request
 
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+@planets_bp.route("", methods=["GET", "POST"])
+def handle_planets():
+    if request.method == "GET":
+        name_query = request.args.get("name")
+        if name_query:
+            planets = Planet.query.filter_by(name=name_query)
+        else:
+            planets = Planet.query.all()
+        planets_response = [planet.to_json() for planet in planets]
+
+        return jsonify(planets_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        new_planet = Planet(
```

Comment:
Consider how this code would behave if "name" or "description" were missing from the request body. One way to handle this sort of invalid request body is with the following code:

```
if "name" not in request_body or "description" not in request_body":
    return {"error":"incomplete request body", 400
```